agent/xds/routes.go

```diff
@@ -129,7 +129,7 @@ func makeUpstreamRouteForDiscoveryChain(
 					}
 					if len(destination.RetryOnStatusCodes) > 0 {
 						if retryPolicy.RetryOn != "" {
-							retryPolicy.RetryOn = ",retriable-status-codes"
+							retryPolicy.RetryOn = retryPolicy.RetryOn + ",retriable-status-codes"
```

Comment:
I guess the original intent was to use `+=`... It's easier to see it's correct with the more explicit concatenation though.